### ISSUE TITLE
[Bug]: A11y: FocusZone should set aria-setsize and aria-posinset to help navigate through the FocusZone

### ISSUE THREAD

**Issue by amylemmon** [microsoft/fluentui#27269]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz
    Memory: 105.15 GB / 127.71 GB
  Browsers:
    Edge: Spartan (44.22621.1413.0), Chromium (110.0.1587.56)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/benlemmon/pen/jOvvXZd

### Bug Description

Open the codepen
With narrator on, tab to the first button.

## Actual Behavior
Narrator announces "Button 1 button"
Visually impaired user is unable to discover that there's more to the list, and would likely press tab skipping some of the UI.

## Expected Behavior
Narrator announces "Button 1 button item 1 of 4" so users know there's more to navigate.

## Note
This can be achieved by adding aria-posinset and aria-setsize to each focusable element, but this should be done by FocusZone rather than pushing developers to do this themselves for the following reasons:

1. We want all scenarios to benefit from this.
2. FocusZone has all the information for this and already does all the heavy lifting logic to focus order
3. Dynamically populated FocusZones may not be able to control how each element is rendered, but FocusZone is already doing DOM manipulation.
4. Each element needs to know how many focusable items there are.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by tudorpopams** [comment]
@smhigley do you feel that this is something feasible? Sounds more like the a11y of a list instead of a focus zone.

**Comment by smhigley** [comment]
This isn't a good idea, `aria-setsize` and `aria-posinset` are only relevant for virtualized content. They will alter the calculated position and set size for an item, but will not affect whether those are read out by screen readers.

This expected result is not achievable, and there is no markup or ARIA that will cause it to happen, short of stuffing it in the accname, which is a pretty severe antipattern: "Narrator announces "Button 1 button item 1 of 4" so users know there's more to navigate."
